### PR TITLE
Assorted Fixes for the `sql` Module

### PR DESCRIPTION
**Pull request by plypaul**
Resolves #

### Description

This PR makes fixes some minor issues found in the `sql` module. Please see view each commit separately.

### CODE REVIEW

File: metricflow/sql/sql_table.py [hunk 1]
```diff
@@ -1,35 +1,16 @@
...
 
-from dbt_semantic_interfaces.implementations.base import (
-    FrozenBaseModel,
-    PydanticCustomInputParser,
-    PydanticParseableValueType,
-)
 
-
-class SqlTable(PydanticCustomInputParser, FrozenBaseModel):
+@dataclass(frozen=True, order=True)
```
Comment (tlento) [inline metricflow/sql/sql_table.py:7]: Oh nice, thanks! There was some reason we couldn't do this originally and I guess we forgot to clean it up later.

File: metricflow/sql/sql_plan.py [hunk 2]
```diff
@@ -42,19 +42,19 @@ def parent_nodes(self) -> List[SqlQueryPlanNode]:  # noqa: D
     @abstractmethod
     def accept(self, visitor: SqlQueryPlanNodeVisitor[VisitorOutputT]) -> VisitorOutputT:
         """Called when a visitor needs to visit this node."""
-        pass
+        raise NotImplementedError
```
Comment (tlento) [inline metricflow/sql/sql_plan.py:45]: This shouldn't matter, but I like the consistency.

**Review by tlento — APPROVED** [review]
raccoon cleaning

File: metricflow/sql/sql_plan.py (hunk 2 above)
Comment (plypaul) [reply metricflow/sql/sql_plan.py:45]: Yeah, it seems like that's what the docs suggest, though I've seen some conflicting examples.

**Pull request merged**